datafusion/core/src/physical_optimizer/dist_enforcement.rs

```diff
@@ -16,10 +16,20 @@
 // under the License.
 
 //! EnforceDistribution optimizer rule inspects the physical plan with respect
-//! to distribution requirements and adds [RepartitionExec]s to satisfy them
-//! when necessary.
+//! to distribution requirements and adds [`RepartitionExec`]s to satisfy them
+//! when necessary. If increasing parallelism is beneficial (and also desirable
+//! according to the configuration), this rule increases partition counts in
+//! the physical plan.
+
+use std::fmt;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
 use crate::config::ConfigOptions;
-use crate::error::Result;
+use crate::datasource::physical_plan::{CsvExec, ParquetExec};
+use crate::error::{DataFusionError, Result};
+use crate::physical_optimizer::sort_enforcement::{unbounded_output, ExecTree};
```

Comment:
It seems like it would be good to pull the things that are now shared between `EnforceSort` and `EnforceDistribution` into a common location might make the dependency structure clearer

`ExecTree` in particular seems like a valuable concept that is not specific to SortEnforcement